src/company_base.h

```diff
@@ -79,6 +79,8 @@ struct CompanyProperties {
 
 	Year inaugurated_year;           ///< Year of starting the company.
 
+	byte rating;                     ///< Ranking nth of the company with 1st being best performing according to performance history.
```

Comment:
You create this variable, but only write and read to it in a single function. That means you don't actually need to allocate the variable here. Saves a bit of memory that is always allocated while it is only used in a very specific case :) But we will see in a bit that I think we don't even need to do this :)